core/src/bson_type_info.rs

```diff
@@ -109,7 +109,28 @@ pub struct BsonTypeInfo {
     pub simple_type_info: Option<SimpleTypeInfo>,
 }
 
```

Comment:
mew rust C doesn't like comparing function pointers.

Reply:
It's a new warning from the newest version of rustc. Comparing function pointers was always UB. We should have removed the derived PartialEQ when we changed to function pointers.